frontends/web/src/routes/buy/pocket.tsx

```diff
@@ -89,19 +93,35 @@ export const Pocket = ({ code }: TProps) => {
     // handle address request from moonpay
     try {
       const message = parseMessage(e.data);
-      if (message.type === V0MessageType.RequestAddress && message.withMessageSignature) {
+      if (message.type === V0MessageType.RequestAddress && message.withMessageSignature && !signing) {
+        signing = true;
+        addressType = message.withScriptType ? String(message.withScriptType) : '';
         signAddress(
-          message.withScriptType ? message.withScriptType : 'p2wpkh',
+          addressType,
           String(message.withMessageSignature),
           code)
           .then(response => {
-            if (response.status === 'abort') {
-              // TODO notify to the widget with a message?
-              alertUser('Message signing aborted by the user');
-            } else {
+            signing = false;
+            if (response.errorCode !== 'userAbort') {
               sendAddress(response.address, response.signature);
             }
-          });
+          })
+          .catch(() => signing = false);
```

Comment:
this is never executed in Qt. if you want it to you have to change `signAddress` to `reject()` upon error, which I think @thisconnect has started doing sometimes. more common in the codebase is to simply check for success inside of `then(...)`.

this and other things should be cleaned up at some point in our error handling - would be nice to unify it at all some point.

Reply:
Didn't know that, thanks! I added it "just in case", now removed since it shouldn't be necessary anyway

Reply:
we are using reject + `catch` in a few places, but at the moment I don't like it as the args received from catch can't be typed. I think "simply check for success inside of then(...)" is better.